gymnasium/envs/registration.py

```diff
@@ -419,6 +423,42 @@ def _check_spec_register(spec: EnvSpec):
         )
 
 
+def find_spec(id: str):
```

Comment:
Note for future travellers: this is an extracted part of the `make` function to avoid code duplication between that and `make_vec`. Its purpose largely overlaps with the `spec` function, and we'll have to disentangle those at some point